src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -45,7 +45,23 @@ private class InterfaceInfo
 
             // Key is implemented member (method, property, or event), value is implementing member (from the 
             // perspective of this type).  Don't allocate until someone needs it.
-            internal ConcurrentDictionary<Symbol, SymbolAndDiagnostics> implementationForInterfaceMemberMap;
+            private ConcurrentDictionary<Symbol, SymbolAndDiagnostics> _implementationForInterfaceMemberMap;
+
+            public ConcurrentDictionary<Symbol, SymbolAndDiagnostics> ImplementationForInterfaceMemberMap
+            {
+                get
+                {
+                    var map = _implementationForInterfaceMemberMap;
```

Comment:
Technically, the ECMA CLI memory model does require a volatile read here, to prevent "speculative" reads from moving before the null check.  But in practice this is not required, as all current supported processors/code generators keep data-dependent reads in order, and we are extremely likely to preserve this on future architectures to avoid breaking code that assumes it.